src/writers/json-writer.cc

```diff
@@ -61,12 +62,21 @@ class JsonWriter : public WriterBase
 			for (unsigned int n = 1; n < file->m_lastLineNr; n++)
 			{
 				IReporter::LineExecutionCount cnt = m_reporter.getLineExecutionCount(file->m_name, n);
+
+				const std::string& line_str = file->m_lineMap[n];
+				const std::string& file_name = it->first;
+				bool should_cover = shouldCover(line_str, file_name);
```

Comment:
I think that all these can probably be moved to reporter.cc instead, to the `lineIsCode` method. I.e., you could do something like `shouldCover(...) && it->second->lineIsCode(...)` there, and then you can probably remove all changes to the writers.

At least it's something to try out.

Reply:
good suggestion, will do